libsolidity/analysis/ControlFlowBuilder.cpp

```diff
@@ -618,6 +672,81 @@ bool ControlFlowBuilder::visitNode(ASTNode const& _node)
 	return true;
 }
 
+void ControlFlowBuilder::checkForReverts(FunctionCall const& _functionCall)
+{
+	auto const& functionType = dynamic_cast<FunctionType const&>(
+		*_functionCall.expression().annotation().type
+	);
+
+	if (!functionType.hasDeclaration())
+		return;
+
+	auto const& unresolvedFunctionDefinition =
+		dynamic_cast<FunctionDefinition const&>(functionType.declaration());
+
+	FunctionDefinition const* functionDefinition = nullptr;
+
+
+	SimpleASTVisitor visitor([&](ASTNode const& _node) {
+		if (auto const* memberAccess = dynamic_cast<MemberAccess const*>(&_node))
+		{
+			if (*memberAccess->annotation().requiredLookup == VirtualLookup::Super)
+			{
+				if (auto const typeType = dynamic_cast<TypeType const*>(memberAccess->expression().annotation().type))
+					if (auto const contractType = dynamic_cast<ContractType const*>(typeType->actualType()))
+					{
+						solAssert(contractType->isSuper(), "");
+						functionDefinition = &unresolvedFunctionDefinition.resolveVirtual(
+							*m_contract,
+							contractType->contractDefinition().superContract(*m_contract)
+						);
+					}
+			}
+			else
+			{
+				solAssert(*memberAccess->annotation().requiredLookup == VirtualLookup::Static, "");
+				functionDefinition = &unresolvedFunctionDefinition;
+			}
+			return false;
+		}
+		else if (auto const* identifier = dynamic_cast<Identifier const*>(&_node))
+		{
+			solAssert(*identifier->annotation().requiredLookup == VirtualLookup::Virtual, "");
+			functionDefinition = &unresolvedFunctionDefinition.resolveVirtual(*m_contract);
+			return false;
+		}
+		return true;
+	},[](ASTNode const&){});
```

Comment:
```suggestion
	}, {});
```

simple brace-initialized `std::function<void(ASTNode const&)>` doesn't work?

Reply:
Nope, that gives me a runtime exception "bad function call"